File: java/arrays/reverse_array.java
public class ReverseArray {
    public static void main(String[] args) {
        // Original array
        int[] arr = {1, 2, 3, 4, 5};

        // Reverse the array using two pointers
        int i = 0, j = arr.length - 1;
        while (i < j) {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
            i++;
            j--;
        }

        // Print reversed array
        System.out.print("Reversed array: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
    }
}

File: python/arrays/reverse_array.py
n = int(input("Enter size of array: "))
arr = list(map(int, input(f"Enter {n} elements: ").split()))

i, j = 0, n - 1
while i < j:
    arr[i], arr[j] = arr[j], arr[i]
    i += 1
    j -= 1

print("Reversed array:", arr)
